backend/inspirehep/serializers.py

```diff
@@ -82,4 +86,18 @@ def serialize_search(self, pid_fetcher, search_result, **kwargs):
             facets. This is used with
             ``inspirehep.search.factories.search.search_factory_only_with_aggs``.
         """
+
+        aggregations = search_result.get("aggregations")
+        new_aggs = deepcopy(aggregations)
+
+        for agg_key, agg_value in aggregations.items():
+            if "nested" in agg_value:
+                nested = agg_value.get("nested")
+                for key in nested:
+                    if key != "doc_count":
+                        new_aggs[key] = nested.get(key)
+                new_aggs.pop(agg_key)
+
+        search_result["aggregations"] = new_aggs
```

Comment:
FYI, I would write this more like follows. First define a helper function:
```python
def unnest_aggregation(agg):
    """Flatten the aggregation dict in case there are nested aggregations.
    
    Warning:
         This mutates the argument.
    """
    if "nested" not in agg:
        return agg
    del agg["nested"]["doc_count"]
    return agg["nested"]
```
(I think it's fine to remove the `doc_count` by mutating the dict here as we won't need the original later, but better be explicit about it).

Then you can do here
```python
aggregations = search_result.get("aggregations", {})
search_result["aggregations"] = {name: unnest_aggregation(agg) for (name, agg) in aggregations.items()}
```
which I believe is more readable and has the benefit of iterating only once over the aggregations instead of twice when using `copy` (or much worse with `deepcopy`).

Reply:
I'm stupid, this doesn't work because you need to unnest twice. Anyway, the general point still stands: it's better to build the resulting `dict` as you go rather than copy the original and mutate it.